examples/src/main/java/io/dapr/examples/workflows/DemoWorkflowClient.java

```diff
@@ -45,22 +36,16 @@ public class DemoWorkflowClient {
    * @throws InterruptedException If program has been interrupted.
    */
   public static void main(String[] args) throws InterruptedException, TimeoutException {
-    DurableTaskClient client = new DurableTaskGrpcClientBuilder().build();
-    try (client) {
-      String name = DemoWorkflow.class.getCanonicalName();
-      String instanceId = client.scheduleNewOrchestrationInstance(name);
-      System.out.printf("Started new orchestration instance: %s%n", instanceId);
-
-      TimeUnit.SECONDS.sleep(5);
+    DurableTaskClient innerClient = new DurableTaskGrpcClientBuilder().build();
+    DaprWorkflowClient client = new DaprWorkflowClient(innerClient);

```

Comment:
We should update this to not take any arguments. This part should be hidden from the user.

Check out how the [ActorClient](https://github.com/mindlessroman/java-sdk/blob/master/sdk-actors/src/main/java/io/dapr/actors/client/ActorClient.java#L48-L69) does it. Ours will be slightly different than this but should follow the same convention of `public Constructor() { this(argsAndOrSettings); }` followed by `private Constructor(Object whateverArgs)`

You may be thinking "But what the heck, the dotnet client takes that arg in the public constructor. why is that okay?" 
 It's a good question. The reason is that the dotnet (framework) has strong dependency injection patterns which allow users to just call `new DaprWorkflowClient()` and dotnet _injects_ the DurableTaskClient _dependency_ for you.

The Java (language) is not so opinionated and so we have to use a different pattern to give the user the same experience. As an aside - if this were the `dapr-springboot-sdk` (framework) we could use DI more liberally; and if we were referencing a `dapr-csharp-sdk` (language) it would likely not use DI at all.